### DISCUSSION TITLE
next build first load JS colors

### DISCUSSION THREAD

**Discussion by daveteu** [vercel/next.js#16647] [Help]
When building, I often see First Load JS color in Red, despite me doing all kind of code optimization.

May I know what are the criterial for the first load JS to be shown as green or yellow instead of red?

Red color signifies critical, or error, and usually a page of red is not a good sign, which requires actions.

So, how are the colors determined? If it's purely base on size, showing it in red is definitely not the best idea as different projects are of different sizes.

![image](https://user-images.githubusercontent.com/18649392/91556217-eec34f00-e964-11ea-9f3a-a292bf4b2ca9.png)

**Comment by ethanryan** [comment]
@daveteu according to [this reddit post](https://www.reddit.com/r/reactjs/comments/hctcmt/in_nextjs_how_do_i_aim_for_green_values_in_first/):

It's just the size of the page.

```
const getPrettySize = (_size: number): string => {
    const size = prettyBytes(_size)
    // green for 0-130kb
    if (_size < 130 * 1000) return chalk.green(size)
    // yellow for 130-170kb
    if (_size < 170 * 1000) return chalk.yellow(size)
    // red for >= 170kb
    return chalk.red.bold(size)
}
```
via the Next.js source: https://github.com/vercel/next.js/blob/279ae19c7e4cb75bd79873b0036332222781b792/packages/next/build/utils.ts#L77-L85

so:

```
< 130kb: green 🟢 
> 130-170kb: yellow 🟡 
> 170kb: red 🔴 
```

**Comment by alamenai** [comment]
For my case, I see all of the in white!

![image](https://github.com/user-attachments/assets/1fab48d8-6516-40a8-ad0b-a53e3de3d465)

**Reply by icyJoseph** [reply]
https://github.com/vercel/next.js/pull/60385

**Reply by alamenai** [reply]
Thank you @icyJoseph  but I'm using the version 14.2.0 and the changes ( colored sizes ) should be appeared,right?

**Reply by icyJoseph** [reply]
The commit, [14b6ae2](https://github.com/vercel/next.js/commit/14b6ae2e707dc99ca19d011d78fccfd4e54b33ae) has been merged to canary there since early v14.0.5-canary.46 :) - first stable release v14.1.0
